src/components/Goodslist.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import '../App.css';
+
+type Props = {
+  goods: string[],
+};
+
+export const Goodslist: React.FC<Props> = ({ goods }) => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortBy, setSortBy] = useState('');
+  const [lengthLimit, setLengthLimit] = useState(1);
+
+  const selectHandler = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setLengthLimit(+event.target.value);
+  };
+
+  const reverse = () => {
+    setIsReversed(prevState => !prevState);
+  };
+
+  const sortByLength = () => {
+    setSortBy('length');
+  };
+
+  const sortByName = () => {
+    setSortBy('name');
+  };
+
+  const reset = () => {
+    setLengthLimit(1);
+    setIsReversed(false);
+    setSortBy('');
+  };
+
+  const selectedGoods = goods.filter(good => {
+    return good.length >= lengthLimit;
+  });
+
+  selectedGoods.sort((good1, good2) => {
+    switch (sortBy) {
+      case 'length':
+        return good1.length - good2.length;
+        break;
+
+      case 'name':
+        return good1.localeCompare(good2);
+        break;
+      default:
+        return 0;
+        break;
+    }
+  });
+
+  if (isReversed) {
+    selectedGoods.reverse();
+  }
+
+  return (
+    <div className="colums">
+      <div className="
+              card
+              column
+              is-half
+              is-offset-one-quarter
+              content is-large
+              "
+      >
+        <h1 className="
+                contnet
+                is-large
+                has-text-centered"
+        >
+          Choise Filter
+        </h1>
+        <div className="column columns">
+          <div className="column">
+            <button
+              className={classNames(
+                'button',
+                'is-success',
+                { 'is-danger': isReversed },
+              )}
+              type="button"
+              onClick={() => reverse()}
```

Comment:
```suggestion
              onClick={reverse}
```
these approaches are equal